imap_processing/idex/idex_packet_parser.py

```diff
@@ -0,0 +1,286 @@
+        )
+        logging.info(
+            "Packet high trigger mode = "
+            + str(pkt.data["IDX__TXHDRHGTRIGMODE"].derived_value)
+        )
+        if pkt.data["IDX__TXHDRLGTRIGMODE"].derived_value != 0:
+            logging.info("Low gain TOF trigger mode enabled.")
+            self.Triggerorigin = "LG"
+        if pkt.data["IDX__TXHDRMGTRIGMODE"].derived_value != 0:
+            logging.info("Mid gain TOF trigger mode enabled.")
+            self.Triggerorigin = "MG"
+        if pkt.data["IDX__TXHDRHGTRIGMODE"].derived_value != 0:
+            logging.info("High gain trigger mode enabled.")
+            self.Triggerorigin = "HG"
+        logging.info(
+            f"AID = {pkt.data['IDX__SCI0AID'].derived_value}"
+        )  # Instrument event number
+        logging.info(
+            f"Event number = {pkt.data['IDX__SCI0EVTNUM'].raw_value}"
+        )  # Event number out of how many events constitute the file
+        logging.info(
+            f"Rice compression enabled = {bool(pkt.data['IDX__SCI0COMP'].raw_value)}"
+        )
+
+
+class IDEXRawDustEvent:
+    def __init__(self, epoch: float):
+        """
+        This function initializes a raw dust event
+
+        Parameters:
+            epoch (float):  Impact time in seconds since January 1st 2012 Midnight UTC
+
+        """
+        self.impact_time = epoch
+        self.TOF_High_bits = ""
+        self.TOF_Mid_bits = ""
+        self.TOF_Low_bits = ""
+        self.Target_Low_bits = ""
+        self.Target_High_bits = ""
+        self.Ion_Grid_bits = ""
+
+    def _parse_high_sample_waveform(self, waveform_raw: str):
+        """
+        Parse a binary string representing a high sample waveform
+        Data arrives in 10 bit chunks
+        """
+        w = bitstring.ConstBitStream(bin=waveform_raw)
+        ints = []
+        while w.pos < len(w):
+            w.read("pad:2")  # skip 2
+            ints += w.readlist(["uint:10"] * 3)
+        return ints
+
+    def _parse_low_sample_waveform(self, waveform_raw: str):
+        """
+        Parse a binary string representing a low sample waveform
+        Data arrives in 12 bit chunks
+        """
+        w = bitstring.ConstBitStream(bin=waveform_raw)
+        ints = []
+        while w.pos < len(w):
+            w.read("pad:8")  # skip 2
+            ints += w.readlist(["uint:12"] * 2)
+        return ints
+
+    def append_raw_data(self, scitype, bits):
+        """
+        This function determines which variable to append the bits
+        to, given a specific scitype.
+        """
+        if scitype == 2:
+            self.TOF_High_bits += bits
+        elif scitype == 4:
+            self.TOF_Low_bits += bits
+        elif scitype == 8:
+            self.TOF_Mid_bits += bits
+        elif scitype == 16:
+            self.Target_Low_bits += bits
+        elif scitype == 32:
+            self.Target_High_bits += bits
+        elif scitype == 64:
+            self.Ion_Grid_bits += bits
```

Comment:
```suggestion
            self.Ion_Grid_bits += bits
        else:
            logging.warning("Unknown science type received: [%s]", scitype)
```
Add some warnings/raises if we are getting into the edge cases?